test/Microsoft.AspNet.Mvc.Core.Test/DefaultActionSelectorTests.cs

```diff
@@ -860,7 +861,7 @@ private static RouteContext CreateRouteContext(string httpMethod)
             var routeData = new RouteData();
             routeData.Routers.Add(new Mock<IRouter>(MockBehavior.Strict).Object);
 
-            var httpContext = new Mock<HttpContext>(MockBehavior.Strict);
+            var httpContext = new Mock<HttpContext>(MockBehavior.Loose);
```

Comment:
This makes me nervous. Can you just set the properties that are being used in the test (assuming it's the service provider).